pkg/cmd/codespace/delete_test.go

```diff
@@ -214,8 +264,20 @@ func TestDelete(t *testing.T) {
 				},
 			}
 			if tt.opts.codespaceName == "" {
-				apiMock.ListCodespacesFunc = func(_ context.Context, _ api.ListCodespacesOptions) ([]*api.Codespace, error) {
-					return tt.codespaces, nil
+				if tt.opts.orgName != "" {
+					apiMock.ListCodespacesFunc = func(_ context.Context, _ api.ListCodespacesOptions) ([]*api.Codespace, error) {
+						var filteredCodespaces []*api.Codespace
+						for _, codespace := range tt.codespaces {
+							if codespace.Owner.Login == "monalisa" {
+								filteredCodespaces = append(filteredCodespaces, codespace)
```

Comment:
You're right that having the stubbed `ListCodespacesFunc` implementation also perform filtering based on username is a smell. I'd say that it doesn't matter what the implementation of `ListCodespacesFunc` is; we just want to assert that it was called with the correct arguments (in this case, the `UserName` field). You can verify that after the logic to be tested finished running: see `apiMock.DeleteCodespaceCalls` in the same test as an example.

The stubbed ListCodespacesFunc can just keep returning a fixed, stubbed set `tt.codespaces` as before.

Reply:
Thanks for the suggestion and the example! I updated the test in this commit https://github.com/cli/cli/pull/6944/commits/bd2c8b67880631af8d8ff80b53237f697daa19b5. I also confirmed that the test indeed fails if I take back my changes 😄